### DISCUSSION TITLE
Static export without clientside JavaScript

### DISCUSSION THREAD

**Discussion by callumlocke** [vercel/next.js#34065] [Ideas]
### Describe the feature you'd like to request

For certain projects, I want the lightness of a pure HTML and CSS static website build, with no clientside JS. These projects can have multiple pages that use shared components (so Next is still a good fit) but the components are all pure and simple so there's no need for clientside hydration.

It's already possible to extend `Document` and omit the `<NextScript />` tag when rendering.  But this has a shortcoming: clientside JS chunks are still included in the build output, and more importantly, they are still downloaded to the client because of `<link rel="preload">` tags rendered by Next's `<Head />` component, wasting bandwidth.

### Describe the solution you'd like

A flag on `next build` (or an alternative command) that tells Next not to generate any clientside script bundles/chunks at all during build. `<NextScript />` would therefore render nothing, and `<Head />` wouldn't render any preload links.

### Describe alternatives you've considered

It looks like it's [possible](https://www.tezify.com/how-to/avoid-preloading-scripts-with-nextjs/) to create your own alternative `Head` component that omits the preload links, and I've tried this, but it feels a bit brittle/unsupported, and clientside chunks are still built unnecessarily. Ideally I'd like a formal way to say "no clientside JS please".

**Comment by stevemarksd** [comment]
That would be amazing.
